### PR TITLE
std: Deprecate 'x'/'X'/'e'/'E' special cases for u8 slices

### PR DESCRIPTION
**Pull request by LemonBoy**
Let's follow the road paved by the removal of 'z'/'Z', the Formatter
pattern is nice enough to let us remove the remaining four special cases
and declare u8 slices free from any special casing!

### CODE REVIEW

**Comment by marler8997** [comment]
Maybe `x` and `X` are common enough to remain?

**Comment by LemonBoy** [comment]
No.

**Comment by LemonBoy** [comment]
Rebased, note that this reverts #8031 as the deprecation warnings were added (and removed) within the same release.

**Review by andrewrk — APPROVED** [review]

File: lib/std/fmt.zig [hunk 1]
```diff
@@ -709,6 +709,87 @@ fn formatFloatValue(
...
+
+/// Return a Formatter for a []const u8 where every byte is formatted as a pair
+/// of lowercase hexadecimal digits.
+pub fn fmtSliceHexLower(bytes: []const u8) std.fmt.Formatter(formatSliceHexLower) {
+    return .{ .data = bytes };
+}
+
+/// Return a Formatter for a []const u8 where every byte is formatted as a pair
+/// of uppercase hexadecimal digits.
+pub fn fmtSliceHexUpper(bytes: []const u8) std.fmt.Formatter(formatSliceHexUpper) {
```
Comment (daurnimator) [inline lib/std/fmt.zig:744]: I don't love that we have different functions for this for upper vs lower case.
Could we instead take the case as an argument?

`pub fn fmtSliceHex(bytes: []const u8, case: enum{Upper, Lower}) std.fmt.Formatter(formatSliceHex) {`

Comment (LemonBoy) [reply lib/std/fmt.zig:744]: No strong opinion on having a single formatter, I went with two separate ones to keep it close to the `x`\`X`.

(IIRC there's a stage1 bug regarding comptime anonymous enum arguments, check out debug.zig)

**Pull request merged**